tools/automate_bindings.py

```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/python
+
+""" Generate bindings of given Wt's class and all its dependencies.
+
+Dependencies are all other classes from interface of methods
+of given class (types of arguments and return values).
+"""
+
+import argparse
+import os
+import re
+
+import pygccxml
+
+TYPE_FROM_LUA_FUNCS = {
+    'int' : 'lua_tointeger',
+    'bool' : 'lua_toboolean',
+}
+
+TYPE_TO_LUA_FUNCS = {
+    'int' : 'lua_pushinteger',
+    'bool' : 'lua_pushboolean',
+}
+
+def parse(filename):
+    # Find out the c++ parser.
+    generator_path, generator_name = pygccxml.utils.find_xml_generator()
+    # Configure the xml generator.
+    xml_generator_config = pygccxml.parser.xml_generator_configuration_t(
+        xml_generator_path = generator_path,
+        xml_generator = generator_name,
+    )
+    # Parse the c++ file.
+    decls = pygccxml.parser.parse([filename], xml_generator_config)
+    # Get access to the global namespace.
+    global_namespace = pygccxml.declarations.get_global_namespace(decls)
+    return global_namespace
+
+def getMethodsAndBases(global_namespace):
+    Wt = global_namespace.namespace('Wt')
+    methods = Wt.member_functions()
+    main_class = Wt.classes()[0]
+    bases = main_class.bases
+    return methods, bases
+
+def getConstructor(global_namespace):
+    Wt = global_namespace.namespace('Wt')
+    main_class = Wt.classes()[0]
+    # TODO (for zer0main).
+    # We need to support multiple constructors so it's just a dummy.
+    return pygccxml.declarations.find_trivial_constructor(main_class)
+
+def getModuleName(filename):
+    return os.path.basename(filename)
+
+INCLUDES_TEMPLATE = r'''
+#include "boost-xtime.hpp"
+#include <Wt/%s>
+
+#include "globals.hpp"
+
+'''
+
+def generateIncludes(module_name):
+    return INCLUDES_TEMPLATE.lstrip() % module_name
+
+def getSelf(module_name):
+    frame = r'''
+    %s* self = luawt_checkFromLua<%s>(L, 1);
+    '''
+    return frame % (module_name, module_name)
+
+def getInbuiltTypeArgument(options):
+    frame = r'''
+    %(argument_type)s %(argument_name)s = %(func)s(L, %(index)s);
+    '''
+    return frame.lstrip() % options
+
+def getComplexArgument(options):
+    frame = r'''
```

Comment:
No. I won't change this code because it  was utterly hard to fix all the indent problems in generated code. lstrip(), indents and stuff must stay the same.